heartbeat/heartbeat.go

```diff
@@ -55,10 +56,11 @@ type Subject struct {
 }
 
 // New creates a new instance of the Heartbeat struct
-func New(hbcfg *config.HeartBeat, replicatorName string, log *logrus.Entry) (*HeartBeat, error) {
+func New(hbcfg *config.HeartBeat, replicatorName string, log *logrus.Entry, hostname ...string) (*HeartBeat, error) {
```

Comment:
hmm, this leaks test implementation details to public API, not sure its great.  

Lets add a function hb.getHostName() and in there you check a local var `stubHostname`, if its empty return that else do `os.Hostname()`, then in your test you can update the stub.  Bit cleaner else this is just weird.